extensions/amp-nested-menu/validator-amp-nested-menu.protoascii

```diff
@@ -0,0 +1,174 @@
+#
+# Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+tags: {  # amp-nested-menu
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-nested-menu"
+    version: "0.1"
+    version: "latest"
+    requires_usage: NONE
```

Comment:
What is the use case where the extension script would be on a page but the tag would not be?

Reply:
Ah yes. One potential use case I can see is `amp-list`. The requirement that the entire nested menu is generated by amp-list via XHR was actually one of the major use cases for this component. Aside from grandfathered components, it looks like the interaction between validator and components used in amp-list-only is not exactly solved. Since we observed a number of websites generating both their nested sidebar and their mega menu via an XHR, I'm inclined to keep this unless there's a better way for the validator to say "this is okay because it will be dynamically generated". I forget which issue it was, but I do recall a thread on this from somewhere...

Reply:
Revisiting this, I realized that given the current model of using this only inside `amp-sidebar`, `amp-sidebar` is **always** responsible for loading the extension, so the developer should not need to manually specify this extension for usage. Removed the allowance for no usage.